### DISCUSSION TITLE
Does Kurtosis pull a container image down each time I run a package?

### DISCUSSION THREAD

**Discussion by guisellaDes** [kurtosis-tech/kurtosis#1274] [Q&A] [answered]
Does Kurtosis pull a container image down each time I run a package?

**Comment by leeederek** [comment, marked as answer]
Hey @guisellaDes : 

Kurtosis will always attempt to pull the latest image from an external registry (e.g. Dockerhub) for each kurtosis run. If the image pull fails and the image exists locally, Kurtosis will use the local image.

Note that if you want to use a private container image in an enclave, the only way you can use that image in Kurtosis is if you build it locally. Kurtosis currently does not support private container registries at this time. Cheers!
